### PR TITLE
closes #466 add log file v4

### LINKED ISSUE #466: Add a "log" file so there'll be an artifact even if the tests are ended early
[Conclusion: Just for v4. Add a "log.txt" file where each report entry is saved as it happens. Upload that as a GitHub artifact at the end.]

That way if they see something failed and want to stop, they'll have something, even if it's slightly more messy. What do folks think? [Do this automatically? Do this only when dev requests it? If the latter, we can't do this for setup and takedown.]

[Include setup and takedown in the logs.]

I wish we could record everything that's logged to the console as well as all our report messages, but I'm not sure we can. See other issues about cucumber's output.

Should we be as verbose as possible in these logs, even logging our own debugging stuff, or should we stick with just stuff that would go in reports?

### PR DESCRIPTION
**Pull request by mcdonaldcarolyn**
Created log.txt and git hub artifact for reports. Closes #466.

### CODE REVIEW

File: lib/scope.js [hunk 1]
```diff
@@ -32,7 +32,10 @@ module.exports = {
     let scenario = await scope.makeSureReportPartsExist( scope );
     scenario.lines.push({ type, value });
 
+    fs.appendFileSync('log.txt', `Report:\n type: ${ type }, value: ${ value }`);
```
Comment (plocket) [inline lib/scope.js:35]: This is my typo from my console log, but it'd be generous of you to fix it - I have a space between the `\n` and "type". This is what I'd love it to be:

```suggestion
    fs.appendFileSync('log.txt', `Report:\ntype: ${ type }, value: ${ value }`);
```

**Review by plocket — CHANGES_REQUESTED** [review]
Great progress! A couple things to add/change that I completely forgot to describe to you. Some of them are nitpicky and if you don't get to them, I'll approve the PR anyway.

When the major items are taken care of, I'll update the changelog because it has a merge conflict from other branches that were recently merged.

File: lib/steps.js [hunk 2]
```diff
@@ -57,6 +57,8 @@ BeforeAll(async () => {
   scope.timeout = 30 * 1000;
   scope.showif_timeout = 600;
   scope.report = new Map();
+  fs.writeFileSync('log.txt', " ");
```
[the remark this answers is not in the document]
Comment (mcdonaldcarolyn) [reply lib/steps.js]: Yes no problem

**Review by plocket — APPROVED** [review]
Looks great! Thank you!

**Pull request merged**